pkg/health/health.go

```diff
@@ -0,0 +1,46 @@
+package health
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spiffe-helper/pkg/sidecar"
+)
+
+type CheckConfig struct {
+	ListenerEnabled bool   `hcl:"listener_enabled"`
+	BindPort        int    `hcl:"bind_port"`
+	HealthPath      string `hcl:"health_path"`
+}
+
+func StartHealthServer(daemonMode *bool, healthCheckConfig CheckConfig, log logrus.FieldLogger, sidecar *sidecar.Sidecar) error {
+	if *daemonMode && healthCheckConfig.ListenerEnabled {
+		http.HandleFunc(healthCheckConfig.HealthPath, func(w http.ResponseWriter, _ *http.Request) {
+			healthy := sidecar.CheckHealth()
+			if healthy {
+				_, err := w.Write([]byte(http.StatusText(http.StatusOK)))
+				log.Error(err)
+				if err != nil {
+					return
+				}
+			} else {
+				statusText := http.StatusText(http.StatusServiceUnavailable)
+				b, err := json.Marshal(sidecar.GetFileWriteStatuses())
```

Comment:
Do you know what this will look like? Do you need the `json:` tags in the struct?

Reply:
It will be serialized with the same names as Go has, unless you put the tags in the struct. So at the moment, it's `{"X509WriteSuccess":true,"JwtWriteSuccesses":{"a.jwt":true,"b.jwt":false}}`. I probably should change that to camel case?

Reply:
I made snake case, since it seemed like that's the pattern SPIRE had.